RUFAS/biophysical/manure/handler.py

```diff
@@ -0,0 +1,281 @@
+
+        info_map_c = {"units": MeasurementUnits.DEGREES_CELSIUS}
+        info_map_m3 = {"units": MeasurementUnits.CUBIC_METERS}
+        cleaning_water_volume = self.determine_cleaning_water_volume_in_main_barn(
+            self.manure_stream.pen_manure_data.num_animals,
+            self.config.cleaning_water_use_rate,
+            self.config.cleaning_water_recycle_fraction,
+        )
+        barn_temperature = self.determine_barn_temperature(conditions.mean_air_temperature)
+
+        total_cleaning_water_volume = (
+            cleaning_water_volume + self.fresh_water_volume_used_for_milking
+        ) * GeneralConstants.LITERS_TO_CUBIC_METERS
+        self._om.add_variable("total_cleaning_water_volume", total_cleaning_water_volume, info_map_m3)
+        self._om.add_variable("barn_temperature", barn_temperature, info_map_c)
+
+        manure_water = self.manure_stream.water + cleaning_water_volume
+
+        manure_total_ammoniacal_nitrogen = max(0.0, self.manure_stream.ammoniacal_nitrogen - self.ammonia_emission)
+        nitrogen = self.manure_stream.nitrogen
+        phosphorus = self.manure_stream.phosphorus
+        potassium = self.manure_stream.potassium
+        ash = self.manure_stream.ash
+        non_degradable_volatile_solids = self.manure_stream.non_degradable_volatile_solids
+        degradable_volatile_solids = self.manure_stream.degradable_volatile_solids
+        volume = self.manure_stream.volume
+        total_solids = self.manure_stream.total_solids
+
+        self.manure_stream = None
+        return {
+            "manure": ManureStream(
+                water=manure_water + total_cleaning_water_volume,
+                ammoniacal_nitrogen=manure_total_ammoniacal_nitrogen,
+                nitrogen=nitrogen,
+                phosphorus=phosphorus,
+                potassium=potassium,
+                ash=ash,
+                non_degradable_volatile_solids=non_degradable_volatile_solids,
+                degradable_volatile_solids=degradable_volatile_solids,
+                volume=volume,
+                total_solids=total_solids,
+                pen_manure_data=None,
+            )
+        }
+
+    @staticmethod
+    def determine_cleaning_water_volume_in_main_barn(
+        num_animals: int, cleaning_water_use_rate: float, cleaning_water_recycle_fraction: float
+    ) -> float:
+        """
+        Calculates the volume of fresh (non-recycled) cleaning water used for, and ultimately added to, a single manure
+         stream on a single simulation day by the manure handler.
+
+        Parameters
+        ----------
+        num_animals : int
+            Number of animals.
+        cleaning_water_use_rate : float
+            The use rate of cleaning water (unitless).
+        cleaning_water_recycle_fraction : float
+            The fraction of cleaning water recycled (unitless).
+
+        Returns
+        -------
+        float
+            The volume of fresh (non-recycled) cleaning water (m^3).
```

Comment:
```suggestion
            The volume of fresh (non-recycled) cleaning water (L).
```
Still in L here, we use this value in the function for `total_cleaning_water_volume` which applies the L to m3 conversion constant. I sort of bungled units relating to these water values as a whole in the design doc though, so that's my bad!